### PR TITLE
BZ2002444: Adds network connectivity requirements

### PR DESCRIPTION
**Pull request by xenolinux**
This PR adds a new section- network connectivity requirements to the IPI section

Version(s): 4.6+ 

Issue: https://bugzilla.redhat.com/show_bug.cgi?id=2002444

Link to docs preview:  https://deploy-preview-45914--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_vsphere/installing-vsphere-installer-provisioned.html#installation-vsphere-installer-network-requirements_installing-vsphere-installer-provisioned

**VSphere:**

http://file.rdu.redhat.com/sdudhgao/BZZ2002444/installing/installing_vsphere/installing-vsphere-installer-provisioned-customizations.html#installation-vsphere-installer-network-requirements_installing-vsphere-installer-provisioned-customizations

http://file.rdu.redhat.com/sdudhgao/BZZ2002444/installing/installing_vsphere/installing-vsphere-installer-provisioned-network-customizations.html#installation-vsphere-installer-network-requirements_installing-vsphere-installer-provisioned-network-customizations

http://file.rdu.redhat.com/sdudhgao/BZZ2002444/installing/installing_vsphere/installing-restricted-networks-installer-provisioned-vsphere.html#installation-vsphere-installer-network-requirements_installing-restricted-networks-installer-provisioned-vsphere

http://file.rdu.redhat.com/sdudhgao/BZZ2002444/installing/installing_vsphere/installing-vsphere-installer-provisioned.html#installation-vsphere-installer-network-requirements_installing-vsphere-installer-provisioned

**VMC**

http://file.rdu.redhat.com/sdudhgao/BZZ2002444/installing/installing_vmc/installing-vmc-customizations.html#installation-vsphere-installer-network-requirements_installing-vmc-customizations

http://file.rdu.redhat.com/sdudhgao/BZZ2002444/installing/installing_vmc/installing-vmc-network-customizations.html#installation-vsphere-installer-network-requirements_installing-vmc-network-customizations

http://file.rdu.redhat.com/sdudhgao/BZZ2002444/installing/installing_vmc/installing-restricted-networks-vmc.html#installation-vsphere-installer-network-requirements_installing-restricted-networks-vmc

QA contact: @tsze-redhat

### CODE REVIEW

**Comment by jinyunma** [comment]
I think that all IPI installation docs for vSphere and VMC need to be updated to append this section - network connectivity requirements

**Comment by xenolinux** [comment]
@jinyunma Could you please check the new change? I added the networking-requirements module in the installing-vmc.adoc assembly.

File: modules/installation-vsphere-installer-network-requirements.adoc [hunk 1]
```diff
@@ -0,0 +1,114 @@
+// Module included in the following assemblies:
+//
+// * installing/installing_vsphere/installing-vsphere-installer-provisioned.adoc
+
+:_content-type: CONCEPT
+[id="installation-vsphere-installer-network-requirements_{context}"]
+= VMware vSphere network connectivity requirements
```
Comment (jinyunma) [inline modules/installation-vsphere-installer-network-requirements.adoc:7]: Suggest to keep consistence as one described in upi section. In upi section, it states "Network connectivity requirements".

File: installing/installing_vmc/installing-vmc.adoc [hunk 2]
```diff
@@ -29,6 +29,8 @@ include::modules/cluster-entitlements.adoc[leveloffset=+1]
 
 include::modules/installation-vsphere-infrastructure.adoc[leveloffset=+1]
 
+include::modules/installation-vsphere-installer-network-requirements.adoc[leveloffset=+1]
```
Comment (jinyunma) [inline installing/installing_vmc/installing-vmc.adoc:32]: thanks for the update, I think that all ipi installation doc need to be updated to add this part, including:
vsphere:
 installing-vsphere-installer-provisioned-customizations
 installing-vsphere-installer-provisioned-network-customization
 installing-restricted-networks-installer-provisioned-vsphere
VMC:
installing-vmc-customizations
installing-vmc-network-customizations
installing-vmc-network-customizations
installing-restricted-networks-vmc

Comment (xenolinux) [reply installing/installing_vmc/installing-vmc.adoc:32]: Addressed

File: installing/installing_vmc/installing-restricted-networks-vmc.adoc [hunk 3]
```diff
@@ -38,6 +38,8 @@ include::modules/cluster-entitlements.adoc[leveloffset=+1]
 
 include::modules/installation-vsphere-infrastructure.adoc[leveloffset=+1]
 
+include::modules/installation-vsphere-installer-network-requirements.adoc[leveloffset=+1]
```
Comment (pneedle-rh) [inline installing/installing_vmc/installing-restricted-networks-vmc.adoc:41]: You could include this module in the same location within each of the assemblies, or as close to the same location as possible. At present, the module positioning does not seem consistent across the assemblies. Introducing consistency in the structure will help with findability.

This applies to all of the assemblies updated in this PR, but I won't mention this comment in all of them.

**Review by pneedle-rh — COMMENTED** [review]
@xenolinux I have completed my review of this PR. Please let me know if you have any questions about my suggestions.

Please also update the commit message for this PR to be more descriptive. At the moment, it is just "BZ".

Thanks!

File: modules/installation-vsphere-installer-network-requirements.adoc [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+//
+// * installing/installing_vsphere/installing-vsphere-installer-provisioned.adoc
+
+:_content-type: CONCEPT
+[id="installation-vsphere-installer-network-requirements_{context}"]
+= Network connectivity requirements
+
+You must configure the network connectivity between machines to allow {product-title} cluster components to communicate. Each machine must resolve the hostnames of all other machines in the cluster.
+
+Review the following details about the required ports for network connectivity.
```
[the remark this answers is not in the document]
Comment (xenolinux) [reply modules/installation-vsphere-installer-network-requirements.adoc:11]: Addressed

File: modules/installation-vsphere-installer-network-requirements.adoc [hunk 5]
```diff
@@ -0,0 +1,112 @@
+// Module included in the following assemblies:
+//
+// * installing/installing_vsphere/installing-vsphere-installer-provisioned.adoc
```
[the remark this answers is not in the document]
Comment (xenolinux) [reply modules/installation-vsphere-installer-network-requirements.adoc:3]: Addressed

File: modules/installation-vsphere-installer-network-requirements.adoc [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+
+Review the following details about the required ports for network connectivity.
+
+ifndef::restricted,origin[]
+[IMPORTANT]
+====
+In connected {product-title} environments, you must connect all nodes to the internet to pull images
+for platform containers and provide telemetry data to Red Hat.
+====
+ifeval::["{context}" == "installing-rhv-restricted-network"]
```
[the remark this answers is not in the document]
Comment (xenolinux) [reply modules/installation-vsphere-installer-network-requirements.adoc:26]: Addressed

File: modules/installation-vsphere-installer-network-requirements.adoc [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+// * installing/installing_vmc/installing-vmc.adoc
+// * installing/installing_vsphere/installing-restricted-networks-installer-provisioned-vsphere.adoc
+// * installing/installing_vsphere/installing-vsphere-installer-provisioned-customizations.adoc
+// * installing/installing_vsphere/installing-vsphere-installer-provisioned-network-customizations.adoc
+
+:_content-type: CONCEPT
+[id="installation-vsphere-installer-network-requirements_{context}"]
+= Network connectivity requirements
+
+You must configure the network connectivity between machines to allow {product-title} cluster components to communicate. Each machine must resolve the hostnames of all other machines in the cluster.
```
Comment (jinyunma) [inline modules/installation-vsphere-installer-network-requirements.adoc:16]: For IPI, the hostname of each machines is resolved through static pod coredns in the cluster, no need additional configuration outside of the cluster, so it's better to remove statement "Each machine must resolve the hostnames of all other machines in the cluster." here

**Comment by jinyunma** [comment]
cc @jcpowermac @rvanderp3 for review if any more comments.

File: modules/installation-vsphere-installer-network-requirements.adoc (hunk 7 above)
Comment (xenolinux) [reply modules/installation-vsphere-installer-network-requirements.adoc]: Addressed

File: installing/installing_vmc/installing-restricted-networks-vmc.adoc (hunk 3 above)
Comment (xenolinux) [reply installing/installing_vmc/installing-restricted-networks-vmc.adoc:41]: Addressed

File: modules/installation-vsphere-installer-network-requirements.adoc [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+
+Review the following details about the required ports for network connectivity.
+
+ifndef::restricted,origin[]
```
[the remark this answers is not in the document]
Comment (xenolinux) [reply modules/installation-vsphere-installer-network-requirements.adoc]: Addressed

**Comment by jcpowermac** [comment]
The scope of the BZ was
```
 - Document required access for: 
             ESXI HOST to Vcenter 
             OCP CLUSTEr to Vcenter 
             Installer host and Vcenter 
```
I don't see that, unless there is a follow up PR?

**Comment by xenolinux** [comment]
> The scope of the BZ was
> 
> ```
>  - Document required access for: 
>              ESXI HOST to Vcenter 
>              OCP CLUSTEr to Vcenter 
>              Installer host and Vcenter 
> ```
> 
> I don't see that, unless there is a follow up PR?

I will create a follow-up PR but I am afraid I don't have sufficient data. Would you be able to help? @jcpowermac

File: modules/installation-vsphere-installer-network-requirements.adoc [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+
+|`10250`-`10259`
+|The default ports that Kubernetes reserves
+
+|`10256`
+|openshift-sdn
+
+.5+|UDP
+|`4789`
+|VXLAN
```
[the remark this answers is not in the document]
Comment (xenolinux) [reply modules/installation-vsphere-installer-network-requirements.adoc]: Good point. Addressed

**Comment by jinyunma** [comment]
> > The scope of the BZ was
> > ```
> >  - Document required access for: 
> >              ESXI HOST to Vcenter 
> >              OCP CLUSTEr to Vcenter 
> >              Installer host and Vcenter 
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I don't see that, unless there is a follow up PR?
> 
> I will create a follow-up PR but I am afraid I don't have sufficient data. Would you be able to help? @jcpowermac

That part has already been described in https://docs.openshift.com/container-platform/4.10/installing/installing_vsphere/installing-vsphere-installer-provisioned.html#prerequisites.

**Comment by xenolinux** [comment]
@jinyunma Is this good to go?

**Comment by bergerhoffer** [comment]
The enterprise-4.12 label has been added to this PR.

This is because your PR targets the `main` branch and is labeled for enterprise-4.11. And any PR going into `main` must also target the latest version branch (enterprise-4.12).

If the update in your PR does NOT apply to version 4.12 onward, please re-target this PR to go directly into the appropriate version branch or branches (enterprise-4.x) instead of main.

**Pull request merged**